File: CodeForces/c++/800 problems/BeautifulMatrix.cpp
#include <iostream>
#include <cstdlib>
#include <cmath>
using namespace std;

int main()
{
    int x;
    int row=0;
    int column=0;
    for(int i=0;i<5;i++)
    {
        for(int j=0; j<5;j++)
        {
            cin >> x;
            if(x==1)
            {
                row=i+1;
                column=j+1;
                break;
            }
        }
        if(row!=0 && column!=0)
            break;
    }
    cout << abs(3 - row) + abs(3 - column) << endl;
    return 0;
}

File: CodeForces/java/src/problems_800/BeautifulMatrix.java
package problems_800;

import java.util.Scanner;

public class BeautifulMatrix {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int row=0;
        int column=0;
        for (int i = 0; i < 5; i++) {

            for (int j = 0; j < 5; j++) {
                int x= sc.nextInt();
                if (x==1){
                    row=j;
                    column=i;
                }
            }
        }

        System.out.println((Math.abs(3-row)+Math.abs(3-column)));
    }
}
